MaxText/layers/attentions.py

```diff
@@ -1858,6 +1860,7 @@ def __call__(
       model_mode: str = common_types.MODEL_MODE_TRAIN,
       deterministic: bool = False,
       previous_chunk: Any = None,
+      slot: Optional[int] = 0,
```

Comment:
Should we default to `None` instead of `0` in the various functions with `slot` as a parameter since `slot=0` is a valid case? It could be inadvertently hiding issues.

Reply:
Updated